src/execute/oas3/parameter-builders.js

```diff
@@ -79,7 +79,7 @@ export function header({ req, parameter, value }) {
     return;
   }
 
-  if (typeof value !== 'undefined') {
+  if (typeof value !== 'undefined' && (!Array.isArray(value) || value.length > 0)) {
```

Comment:
Can we make it a bit clearer? Now I can read it as sentence, instead of analyzing the boolean logic.

```suggestion
  if (typeof value !== 'undefined' || (Array.isArray(value) && value.length > 0)) {
```